### PR TITLE
NPM packages for react and node

### PR DESCRIPTION
**Pull request by Lucasarenaterry**
Screenshot 2024-03-06 at 18 29 18
Screenshot 2024-03-07 at 10 47 43
Screenshot 2024-03-07 at 10 48 06

### CODE REVIEW

File: packages/node/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+npm install @adcaptcha/node
+```
+
+## Usage
+
+```jsx
+import { verify } from '@adcaptcha/react';
+
+try {
+  const response = await verify(apiKey, token)
```
Comment (orssich) [inline packages/node/README.md:29]: Could do with more docs on what is in the return object

File: packages/node/src/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export { verify } from './verify';
```
Comment (orssich) [inline packages/node/src/index.ts]: new line

File: packages/node/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    "outDir": "./dist",                       
+    "strict": true,                           
+    "esModuleInterop": true,                 
+    "skipLibCheck": true,
+    "noImplicitAny": false,                   
+    "forceConsistentCasingInFileNames": true 
+  },
+  "include": ["."],
+  "exclude": ["dist", "node_modules"]
+}
```
Comment (orssich) [inline packages/node/tsconfig.json]: new line

File: packages/react/.babelrc [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+    "presets": ["@babel/preset-env", "@babel/preset-typescript", "@babel/preset-react"]
+}
```
Comment (orssich) [inline packages/react/.babelrc]: new line

File: packages/react/jest.config.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+    testEnvironment: 'jsdom'
+};
```
Comment (orssich) [inline packages/react/jest.config.js]: new line

File: packages/react/src/util.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    script.type = 'module';
+    script.async = true;
+    script.onload = function () {
+      window.adcap.init();
+      resolve();
+    };
+    script.onerror = reject;
+    document.getElementsByTagName('head')[0].appendChild(script);
+  });
+}
```
Comment (orssich) [inline packages/react/src/util.ts]: new line

File: packages/react/test/adCAPTCHA.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    const placementID = 'test-placement-id';
+    render(<AdCAPTCHA placementID={placementID} />);
+
+    await waitFor(() => {
+      const adCaptchaElement = screen.getByTestId('adCaptcha');
+      expect(adCaptchaElement).toBeDefined();
+      expect(adCaptchaElement.getAttribute('data-adcaptcha')).toEqual(placementID);
+    });
+  });
+});
```
Comment (orssich) [inline packages/react/test/adCAPTCHA.spec.tsx:21]: new line

File: packages/react/tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    "moduleResolution": "Bundler",
+    "declaration": true,
+    "outDir": "dist",
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "strict": true,
+    "skipLibCheck": true
+  },
+  "exclude": ["dist", "node_modules"]
+}
```
Comment (orssich) [inline packages/react/tsconfig.json]: new line

File: packages/react/tsup.config.ts [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+export default defineConfig({
+  entry: {
+    index: './src/index.ts'
+  },
+  external: ['react'],
+  format: ['cjs'],
+  dts: true,
+  clean: true,
+  skipNodeModulesBundle: true
+});
```
Comment (orssich) [inline packages/react/tsup.config.ts]: new line

File: packages/react/README.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
+# adCAPTCHA React
```
Comment (orssich) [inline packages/react/README.md:1]: To the uninitiated it's not really clear what this component is doing - some explanation would be good

File: packages/node/README.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
+# adCAPTCHA React
+
+This package provides a Node component for adCAPTCHA.
```
Comment (orssich) [inline packages/node/README.md]: More explanation of what this is actually providing. If it's verifying  success token then explain that

File: packages/node/README.md (hunk 1 above)
Comment (orssich) [reply packages/node/README.md:29]: What is `apiKey`, where does it come from? tell the reader! Perhaps give example

Same for `token`

**Review by orssich — CHANGES_REQUESTED** [review]

File: packages/react/README.md (hunk 10 above)
Comment (orssich) [reply packages/react/README.md:1]: I still think you need to clarify what is happening here. This component does all the following:

- Show a captcha
- on success it gets a token (tell them what to with with it)

Some other things worth covering/examples:

- what happens on a failure
- callback examples as mentioned [here](https://docs.adcaptcha.com/callbacks)

Put yourself in the shoes of a rushed developer who just wants clear instructions/examples of what to do so they can just copy paste it

**Review by orssich — CHANGES_REQUESTED** [review]

**Review by MattGibney — APPROVED** [review]

**Review by orssich — APPROVED** [review]

**Pull request merged**